pkg/ddl/job_submitter.go

```diff
@@ -579,13 +594,23 @@ func assignGIDsForJobs(jobWs []*JobWrapper, ids []int64) {
 			pInfo.NewTableID = pInfo.Definitions[0].ID
 		case model.ActionTruncateTable:
 			if !jobW.IDAllocated {
-				jobW.Args[0] = alloc.next()
-				partCount := jobW.Args[3].(int)
-				partIDs := make([]int64, partCount)
-				for i := range partIDs {
-					partIDs[i] = alloc.next()
+				if jobW.Version == model.JobVersion1 {
```

Comment:
for these getter logic, I prefer we always use a function to convert all version representations to `TruncateTableArgs`, then caller use `TruncateTableArgs`.

like
 
```golang
if !jobW.IDAllocated {
  arg := getTruncateTableArgs(job)
  arg.NewTableID = alloc.next()
  ...
```

Reply:
for new jobs, i plan put JobArgs into JobWrapper, and fill to job on insert, so we can use it directly here.